### PR TITLE
Assigning "Change submissions" implies "Add submissions"

### LINKED ISSUE #3148: Review permissions for duplicate submission feature
## Description

Currently, users need `Edit submissions` to be able to duplicate a submission. When writing the unit tests for row level permissions I wondered whether it would be better to assign `Add submissions` because duplicating a submission is like adding a new submission. 

But @jnm said:
> One wrinkle is that the use case for duplicating submissions assumes the duplicate will usually be edited immediately after creation

That means the owner/ the managers should assign both permissions. It would be make sense, but as @tinok said
> from a UX side I think it's easier to assign one permission instead of two.

If we decide to use both, we would need to document it. Front-end code should be updated too.

related to #2768

### LINKED ISSUE #3218: Users cannot edit submissions with "Change submissions" permission only when project requires authentication
## Description

When users receive "Change submissions" permission, they cannot edit submissions if the owner requires authentication on the project. They need "Add submissions" permission as well to be able to save the edit submission(s).

## Expected behavior
Enketo should return a successful message

## Actual behavior 

Enketo returns an 403 error

### PR DESCRIPTION
**Pull request by noliveleger**
## Checklist

1. [x] If you've added code that should be tested, add tests
2. [x] If you've changed APIs, update (or create!) the documentation
3. [x] Ensure the tests pass
4. [x] Make sure that your code lints and that you've followed [our coding style](https://github.com/kobotoolbox/kpi/blob/master/CONTRIBUTING.md)
5. [x] Write a description of your work suitable for publishing on [our forum](https://community.kobotoolbox.org/tag/release-notes)
6. [x] Mention any related issues in this repository (as #ISSUE) and in other repositories (as kobotoolbox/other#ISSUE)

## Description

Assigning "Change submissions" implies "Add submissions"

## Related issues

Fixes #3218
Closes #3148

### CODE REVIEW

**Comment by noliveleger** [comment]
@joshuaberetta, I'm sorry I made you change things in kpi#3157 to revert (part of) them back in this PR.

File: jsapp/js/components/permissions/userAssetPermsEditor.es6 [hunk 1]
```diff
@@ -160,6 +160,11 @@ class UserAssetPermsEditor extends React.Component {
       stateObj.formViewDisabled = true;
     }
 
+    // checking `submissionsEdit` implies having `submissionsAdd` checked
+    if (stateObj.submissionsEdit) {
+      stateObj.submissionsAdd = true;
+    }
```
Comment (joshuaberetta) [inline jsapp/js/components/permissions/userAssetPermsEditor.es6:166]: The "Add submissions" checkbox should also be greyed out like the other options when "Change submissions" is ticked.

Screen Shot 2021-05-17 at 4 23 20 PM

**Review by joshuaberetta — CHANGES_REQUESTED** [review]

File: jsapp/js/components/permissions/userAssetPermsEditor.es6 (hunk 1 above)
Comment (joshuaberetta) [reply jsapp/js/components/permissions/userAssetPermsEditor.es6:166]: Looks like this does the trick:

```diff
diff --git a/jsapp/js/components/permissions/userAssetPermsEditor.es6 b/jsapp/js/components/permissions/userAssetPermsEditor.es6
index 7f231c2a5..032079252 100644
--- a/jsapp/js/components/permissions/userAssetPermsEditor.es6
+++ b/jsapp/js/components/permissions/userAssetPermsEditor.es6
@@ -163,6 +163,7 @@ class UserAssetPermsEditor extends React.Component {
     // checking `submissionsEdit` implies having `submissionsAdd` checked
     if (stateObj.submissionsEdit) {
       stateObj.submissionsAdd = true;
+      stateObj.submissionsAddDisabled = true;
     }

     // checking these options implies having `submissionsView` checked
```

Comment (noliveleger) [reply jsapp/js/components/permissions/userAssetPermsEditor.es6:166]: 👍 
Nice catch ;-)

**Review by joshuaberetta — APPROVED** [review]

**Pull request merged**